api/v1beta1/azuremachine_webhook.go

```diff
@@ -17,20 +17,42 @@ limitations under the License.
 package v1beta1
 
 import (
+	"context"
 	"reflect"
 
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	webhookutils "sigs.k8s.io/cluster-api-provider-azure/util/webhook"
+	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
 )
 
+// SetupAzureMachineWebhookWithManager sets up and registers the webhook with the manager.
+func SetupAzureMachineWebhookWithManager(mgr ctrl.Manager) error {
+	mw := &azureMachineWebhook{Client: mgr.GetClient()}
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(&AzureMachine{}).
+		WithDefaulter(mw).
+		WithValidator(mw).
+		Complete()
+}
+
 // +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1beta1-azuremachine,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=azuremachines,versions=v1beta1,name=validation.azuremachine.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
 // +kubebuilder:webhook:verbs=create;update,path=/mutate-infrastructure-cluster-x-k8s-io-v1beta1-azuremachine,mutating=true,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=azuremachines,versions=v1beta1,name=default.azuremachine.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
 
+// azureMachineWebhook implements a validating and defaulting webhook for AzureMachines.
+type azureMachineWebhook struct {
+	Client client.Client
+}
+
+var _ webhook.CustomDefaulter = &azureMachineWebhook{}
+var _ webhook.CustomValidator = &azureMachineWebhook{}
```

Comment:
Ah that makes sense. Do you mean that it's not necessary to include these lines because the calls to `WithDefaulter` and `WithValidator` accomplish the same thing?